app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -84,6 +90,28 @@ class HomeFragment : Fragment() {
         settingsButton.setOnClickListener { v ->
             onSettingsPressed?.invoke()
         }
+
+        registerForContextMenu(view)
+    }
+
+    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
+        super.onCreateContextMenu(menu, v, menuInfo)
+        activity.menuInflater.inflate(R.menu.menu_context_hometile, menu)
+    }
+
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.remove -> {
+                val tileToRemove = homeTileAdapter.homeTiles[getFocusedTilePosition()]
+                if (tileToRemove is BundledHomeTile) {
+                    BundledTilesManager.getInstance(context).unpinSite(context, tileToRemove.url.toUri()!!)
```

Comment:
You can't pass in a null uri here - if you look at the code in unpinSite, that's not handled. One thing you can do is check to make sure the url isn't null before passing it around.

Before you !! things (which you should do ONLY when you are absolutely sure that it's not null). I don't see a comment for why you're sure this isn't null, or (even better and preferred) a null check, so this seems like an unjustified !!.

Always be suspicious of your inputs, and don't assume that they aren't null because that causes crashes.